Library/Homebrew/sandbox.rb

```diff
@@ -33,6 +23,23 @@ def allow_write(path, options={})
                       :filter => filter
   end
 
+  def allow_write_temp_and_cache
+    allow_write "/private/tmp", :type => :subpath
+    allow_write "/private/var/folders", :type => :subpath
+    allow_write HOMEBREW_TEMP, :type => :subpath
+    allow_write HOMEBREW_CACHE, :type => :subpath
+  end
+
+  def allow_write_cellar(formula)
+    allow_write formula.rack, :type => :subpath
+    allow_write formula.etc, :type => :subpath
+    allow_write formula.var, :type => :subpath
+  end
+
+  def allow_write_log(formula)
+    allow_write HOMEBREW_LOGS/formula.name, :type => :subpath
+  end
```

Comment:
* `allow_write_temp_and_cache` is still required, it won't be used in the test case. 
* Maybe it's a little overkill, but I think it's fine as when we need other type, this can help to reduce the internal duplicated code. Also, I'm considering to change `allow_write "/private/var/folders", :type => :subpath` to a more strict regex. But I need to do more research on this first.